crates/turbo-tasks-macros/src/func.rs

```diff
@@ -154,9 +154,25 @@ pub fn gen_native_function_code(
         }
     }
     let original_call_code = if async_function {
-        quote! { #original_function(#(#input_arguments),*).await }
+        quote! {
+            turbo_tasks::macro_helpers::tracing::Instrument::instrument(
+                async {
+                    let turbo_tasks_result = #original_function(#(#input_arguments),*).await;
+                    turbo_tasks::macro_helpers::notify_scheduled_tasks();
+                    turbo_tasks_result
+                },
+                turbo_tasks::macro_helpers::tracing::info_span!(#name_code)
+            ).await
```

Comment:
Can we expose a turbo tasks API for this instead of doing this in the macro itself? We should strive to keep the macro output minimal.

e.g. `turbo_tasks::macro_helpers::instrument!(#name_code, #original_function(#(#input_arguments),*))`